### ISSUE TITLE
Instant loading: browser back button breaks internal links

### ISSUE THREAD

**Issue by thesuperzapper** [squidfunk/mkdocs-material#6345]
### Bug description

On all browsers, if you have instant loading enabled, and click on an internal link, then use the browser's back button, attempting to use the same link again will do nothing.

### Related links

- https://github.com/squidfunk/mkdocs-material/issues/5023

### Reproduction

[9.4.8-back-button-breaks-links.zip](https://github.com/squidfunk/mkdocs-material/files/13370947/9.4.8-back-button-breaks-links.zip)

### Steps to reproduce

1. Run any site with instant loading
2. Click on an internal link
3. Use browser's back button
4. Try click the same internal link again
5. Notice that clicking the link does nothing

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. We'll fix this once we find the time to work on instant loading again.

**Comment by thesuperzapper** [comment]
@squidfunk I appreciate it, I really like the instant loading feature because it prevents the jiggling on every page navigation that you get if you turn it off (which I find distracting).

**Comment by squidfunk** [comment]
I'm glad you like it! My time is currently bound with other parts of the theme + community support. I really hope for more sponsors to support the project, so I can delegate more of the community work to other folks, fixing bugs and churning out new features faster ☺️ But you know how it is, most users think Open Source is free, whereas it's always someone else's time.

**Comment by thesuperzapper** [comment]
@squidfunk I am intimately aware, and very grateful for your time, and I hope to sponsor once my project (deployKF) starts making money!

**Comment by thesuperzapper** [comment]
@squidfunk I have tracked down the cause of this issue, the issue no longer appears when I remove the following line:

https://github.com/squidfunk/mkdocs-material/blob/c0aca213603644f80d8137e0932ad684f97d32e1/src/templates/assets/javascripts/integrations/instant/index.ts#L413-L415

I think it's worth just removing this line and dealing with the fact that crazy users can add hundreds of history elements if they spam-click stuff.

But there is probably also some other fix that only prevents repeated anchor clicks and not actual page/path moves.

**Comment by squidfunk** [comment]
Thanks for investigating. We want to solve this thoroughly, so just removing this line is not that simple. We added this line for reason. We need to refactor the logic, and I need to set aside time for that. Will come back to this in the next months.

**Comment by squidfunk** [comment]
Nope, sorry. The introduction of additional feature flags needs to be done very carefully. You know, this project is used by tens of thousands of projects, including thousands of companies. We need to keep it stable. In the meantime, please disable instant loading to mitigate the problem. We'll work on that very soon again. We also hope for more funding from companies, which will allow us to fix problems and push out new features faster. In the meantime, we need to budget our, and especially my, time. I'm sorry for the inconvenience this is causing you or your company, but you know how it is.

**Comment by thesuperzapper** [comment]
I totally understand thanks for all your hard work! I have made a [temporary fork](https://github.com/deployKF/mkdocs-material) with that line removed so Instant navigation works properly on my site.

**Comment by squidfunk** [comment]
Fixed in https://github.com/squidfunk/mkdocs-material-insiders/pull/79 – the fixes are currently only in Insiders, so anybody having access can already test them. I still need to do some minor refactoring that shouldn't impact the functionality, and after that, I'll backport the fixes to the community edition. I just wanted to give them into the hands of users as soon as possible. This should resolve all known problems with instant loading, at least I've tested this and all other issues.
